### ISSUE TITLE
Bug in MoveGen

### ISSUE THREAD

**Issue by s-arash** [jordanbray/chess#15]
There appears to be a bug in move generation. This piece of code panics:
```Rust
let board = Board::from_fen("rnbqkbnr/ppp2pp1/4p3/3N4/3PpPp1/8/PPP3PP/R1B1KBNR b KQkq f3 0 1".to_string()).unwrap();

let move_gen = MoveGen::new_legal(&board);

```

**Comment by jordanbray** [comment]
I probably deserve a bug with how much stuff I changed.  I'll look at this in more detail in a few hours when I get home from work.

Thanks

**Comment by jordanbray** [comment]
Fixed in 3.0.1.
